src/core/qgsapplication.cpp

```diff
@@ -785,6 +785,11 @@ QIcon QgsApplication::getThemeIcon( const QString &name, const QColor &fillColor
     const QByteArray svgContent = QgsApplication::svgCache()->svgContent( path, 16, fillColor, strokeColor, 1, 1 );
 
     const QString iconPath = sIconCacheDir()->filePath( cacheKey + QStringLiteral( ".svg" ) );
+    if ( QDir dir = QFileInfo( iconPath ).dir(); !dir.exists() )
+    {
+      dir.mkpath( "." );
+    }
```

Comment:
Can you pull this out to a separate pull request? Its unrelated to the legend changes.

Reply:
Pull request #56171 has now been submitted. It will have to be accepted first so I can then rebase this branch and continue working on this pull request.